apps/backend/src/deposits/cash-deposit.service.ts

```diff
@@ -34,12 +35,16 @@ export class CashDepositService {
    * @returns
    * @description Create a new cash deposit
    */
-  async createCashDeposit(data: CashDepositEntity): Promise<CashDepositEntity> {
+
+  async saveCashDepositEntities(
+    data: CashDepositEntity[]
+  ): Promise<CashDepositEntity[]> {
     try {
-      return await this.cashDepositRepo.save(this.cashDepositRepo.create(data));
-    } catch (err) {
-      this.appLogger.error(err);
-      throw err;
+      const entities = data.map((d) => this.cashDepositRepo.create(d));
+      return mapLimit(entities, (entity) => this.cashDepositRepo.save(entity));
```

Comment:
I think we could also use forEach here and not return anything (we're not using the returned values). I believe forEach is about 30% faster?

Reply:
we would still need the map limit to throttle calls to db.